src/typechecker.c

```diff
@@ -200,10 +200,10 @@ NODISCARD static Type *common_type(Type *a, Type *b) {
 }
 
 /// Check if a type is a pointer type.
-NODISCARD static bool is_pointer(Type *type) { return type->kind == TYPE_POINTER; }
+NODISCARD static bool is_pointer(Type *type) { return type_is_pointer(type); }
 
 /// Check if a type is an array type.
-NODISCARD static bool is_array(Type *type) { return type->kind == TYPE_ARRAY; }
+NODISCARD static bool is_array(Type *type) { return type_is_array(type); }
```

Comment:
I feel like these are a bit unnecessary now and should just be replaced w/ `type_is_pointer()` and `type_is_array()`... I think it might also make sense to slap `NODISCARD` on all the `type_is_...` functions in ast.h, since it doesn’t really make sense to call e.g. `type_is_pointer()` and then not use the result. 

Also, just in case you don’t know, `NODISCARD` only has to be in the declaration, not the definition (i.e. you’d only need to add it in ast.h, not ast.c).

Reply:
8282edc50843fc5ec882469198e1aedb149dd46e

Done here. I made a bunch of things in `ast.h` nodiscard. This makes me want to have nodiscard be the default in this compiler's lang, and have a `discardable` attribute

Reply:
> This makes me want to have nodiscard be the default

You and a lot of C++ programmers haha

Reply:
I see; I'll add this to the TODO, then :Þ